ray-project/ray
# Update Dashboard UI and grafana dashboard to scope things to a single ray session

Utilizes the SessionName tag in all the emitted metrics. Update api/snapshot endpoint to use a cached sessionNamevariable instead of fetching from internal_kv every time.

Manual test step:
1. Launch a ray cluster
2. Run a workload
3. View metrics and progress bars
4. Restart ray
5. Refresh dashboard, see data disappears
6. Run a workload
7. See progress bar only shows data about the job 01000000 in the current ray session

Signed-off-by: Alan Guo <aguo@anyscale.com>





## Why are these changes needed?



## Related issue number



## Checks

- Testing Strategy
